pandas/tests/frame/test_constructors.py

```diff
@@ -2869,6 +2869,7 @@ def test_from_tzaware_mixed_object_array(self):
         ]
         assert (res.dtypes == expected_dtypes).all()
 
+    @pytest.mark.xfail(reason="DatetimeArray._from_sequence no longer accepts i8")
```

Comment:
we need a note for this?

Reply:
the question is if we want to disable/deprecate this or find a way to make it work

Reply:
is this something we want to support or disallow?

Reply:
To be clear, the question is not so much about `_from_sequence`, but rather about our public constructors (`Series()`, `Index()`, `array()`).  
Currently, those support converting integers to datetime dtype. For example:

```
pd.Series([1, 2, 3], dtype="datetime64[ns]")
```

works fine (and the same for the other two). 
So the question is if we want to keep this kind of behaviour. Personally I would say there is not much harm in allowing it (also given that eg numpy and pyarrow have the same behaviour), it's only when there are timezones that there could be a bit more ambiguity potentially (but eg for timedelta that's already not relevant).

Reply:
I tend to agree.  Any thoughts on where to catch this case?

Reply:
>  Any thoughts on where to catch this case?

Not fully sure what you mean, but (until https://github.com/pandas-dev/pandas/issues/33254 is resolved) doing such conversions is the responsibility of `_from_sequence` (for `pd.array`), so if we want to keep this capability, at the moment it is `_from_sequence` that needs to accept ints.

Reply:
> Not fully sure what you mean

I mean that we could make this Series construction work without loosening the _from_sequence restrictions by and instead calling [??] in this case.

> so if we want to keep this capability, at the moment it is _from_sequence that needs to accept ints.

To be clear, is this what you are suggesting we do?  My impression from previous conversations is that you are unambiguously in favor of this tightening.

Reply:
My earlier suggestion at https://github.com/pandas-dev/pandas/pull/37179#issuecomment-713121153 was to use `_from_sequence`/`_from_sequence_strict` (instead of `_from_sequence_non_strict`/`_from_sequence`), until we decide on better naming for this. So keeping the non-strict behaviour of `_from_sequence` for now. 

> My impression from previous conversations is that you are unambiguously in favor of this tightening.

Yes, for sure. But before we can implement that, we still need to come up with an API for the non-strict conversion as well, as that is still something we need to support. That's the whole discussion of #33254